Core/Src/bms.c

```diff
@@ -307,3 +308,187 @@ void updateTractiveCurrent()
+			{
+				case GCAN_SEGMENT_1:
+				case GCAN_SEGMENT_2:
+				case GCAN_SEGMENT_3:
+				case GCAN_SEGMENT_4:
+				case GCAN_SEGMENT_5:
+				case GCAN_SEGMENT_6:
+				case GCAN_SEGMENT_7:
+					for (int32_t i = 0; i < 4; i++)
+					{
+						update_and_queue_param_float(cellVoltageStatsParams[gcanUpdateState][0], gBms.bmb[gcanUpdateState].stackV);
+						update_and_queue_param_float(cellVoltageStatsParams[gcanUpdateState][1], gBms.bmb[gcanUpdateState].avgBrickV);
+						update_and_queue_param_float(cellVoltageStatsParams[gcanUpdateState][2], gBms.bmb[gcanUpdateState].maxBrickV);
+						update_and_queue_param_float(cellVoltageStatsParams[gcanUpdateState][3], gBms.bmb[gcanUpdateState].minBrickV);
+					}
+
+					for (int32_t i = 0; i < NUM_BRICKS_PER_BMB; i++)
+					{
+						update_and_queue_param_float(cellVoltageParams[gcanUpdateState][i], gBms.bmb[gcanUpdateState].brickV[i]);
+					}
+
+					for (int32_t i = 0; i < NUM_BRICKS_PER_BMB; i++)
+					{
+						update_and_queue_param_float(cellTempParams[gcanUpdateState][i], gBms.bmb[gcanUpdateState].brickTemp[i]);
+					}
+
+					for (int32_t i = 0; i < NUM_BOARD_TEMP_PER_BMB; i++)
+					{
+						update_and_queue_param_float(boardTempParams[gcanUpdateState][i], gBms.bmb[gcanUpdateState].boardTemp[i]);
+					}
+					break;
+				
+				case GCAN_CELL_TEMP_STATS:
+					for (int32_t i = 0; i < NUM_BMBS_IN_ACCUMULATOR; i++)
+					{
+						update_and_queue_param_float(cellTempStatsParams[i][0], gBms.bmb[i].avgBrickTemp);
+						update_and_queue_param_float(cellTempStatsParams[i][1], gBms.bmb[i].maxBrickTemp);
+						update_and_queue_param_float(cellTempStatsParams[i][2], gBms.bmb[i].minBrickTemp);
+					}
+
+					for (int32_t i = 0; i < NUM_BRICKS_PER_BMB; i++)
+					{
+						update_and_queue_param_u8(balswenParams[0][i], gBms.bmb[0].balSwEnabled[i]);
+					}
+
+					// TODO Populate fault data
+					update_and_queue_param_u8(&amsFault_state, 0);
+					update_and_queue_param_u8(&bspdFault_state, 0);
+
+					break;
+				case GCAN_BOARD_TEMP_STATS:
+					for (int32_t i = 0; i < NUM_BMBS_IN_ACCUMULATOR; i++)
+					{
+						update_and_queue_param_float(boardTempStatsParams[i][0], gBms.bmb[i].avgBoardTemp);
+						update_and_queue_param_float(boardTempStatsParams[i][1], gBms.bmb[i].maxBoardTemp);
+						update_and_queue_param_float(boardTempStatsParams[i][2], gBms.bmb[i].minBoardTemp);
+					}
+
+					for (int32_t i = 0; i < NUM_BRICKS_PER_BMB; i++)
+					{
+						update_and_queue_param_u8(balswenParams[1][i], gBms.bmb[1].balSwEnabled[i]);
+					}
+
+					// TODO Populate fault data
+					update_and_queue_param_u8(&imdFault_state, 0);
+					update_and_queue_param_u8(&imdFaultInfo_state, 0);
+					
+					break;
+				case GCAN_BALSWEN:
+					for (int32_t i = 2; i < NUM_BMBS_IN_ACCUMULATOR; i++)
+					{
+						for (int32_t j = 0; j < NUM_BRICKS_PER_BMB; j++)
+						{
+							update_and_queue_param_u8(balswenParams[i][j], gBms.bmb[i].balSwEnabled[j]);
+						}
+					}
+
+					//TODO Add sensor status bytes
```

Comment:
I still want to add gcan variables for the sensor status. But we should talk about how to do that best. Do you like just sending the index of the first cell with an error?